### PR TITLE
Validate loadout subclass config fragment count

### LINKED ISSUE #8303: Loadouts / Loadout Optimizer mostly don't check Subclass Fragment capacity
- [x] It's possible to have a subclass in-game with aspects allowing 3 fragments but 4 fragments equipped. The fourth fragment is simply disabled and has no effect, but creating a Loadout from equipped takes those 4 fragments even though it's not a valid setup.
- [ ] Even with a valid Loadout subclass setup, it's trivial to switch from 4 fragment capacity to 3 without fixing up fragments and end up with an invalid setup.
- [ ] Loadouts and LO will treat stats from excess fragments as active without any indication that the setup is invalid. Contrary to armor mods, which have the mod assignment drawer listing unsassigned mods, and LO refusing to show armor sets if the mods don't fit, subclass config has no indication in the Loadout drawer or LO that the capacity isn't there, so Loadouts and LO will calculate with stats the user considers wrong.

Creating socket overrides from the equipped subclass should check energy capacity, Loadouts and LO should indicate when fragments are invalid and not honor their stats.

(The only place where capacity is being checked is in the Subclass Plug drawer, so that should be extended.)

Edit: Actually, maybe silently eating stats isn't great either... probably just needs a better indication.

### PR DESCRIPTION
**Pull request by robojumper**
"Fill in using equipped" now only copies active fragments. The main loadouts view, the loadout edit view, and loadout optimizer all indicate when there are more fragments than what the aspects grant

grafik

Fixes #8303.

### CODE REVIEW

File: src/app/loadout-drawer/loadout-utils.ts [hunk 1]
```diff
@@ -90,7 +96,8 @@ export function createSocketOverridesFromEquipped(item: DimItem) {
           // Only save them if they're valid plug options though, otherwise
           // we'd save the empty stasis sockets that Void 3.0 spawns with
           plugFitsIntoSocket(socket, socket.plugged.plugDef.hash) &&
-          socket.plugged.plugDef.hash !== socket.emptyPlugItemHash
+          socket.plugged.plugDef.hash !== socket.emptyPlugItemHash &&
+          (category.category.hash !== SocketCategoryHashes.Fragments || activeFragments-- > 0)
```
Comment (robojumper) [inline src/app/loadout-drawer/loadout-utils.ts:100]: Let me know if the short circuit + postdecrement is too clever.

Comment (nev-r) [reply src/app/loadout-drawer/loadout-utils.ts:100]: even if it's the right amount of clever it's extremely missable and would benefit from a comment

**Review by bhollis — APPROVED** [review]

**Comment by robojumper** [comment]
This has some overlap with #8348 and if that PR properly disables inactive plugs, we could simply trust the `ResolvedLoadoutItem`'s sockets instead of having to inspect the socketOverrides... holding off with this for now.

**Comment by robojumper** [comment]
Closing because this needs design decisions first -- https://github.com/DestinyItemManager/DIM/issues/8303#issuecomment-1203689169, will revive the `createSocketOverridesFromEquipped` part separately.

**Pull request closed without merging**